tools/projmgr/src/ProjMgrYamlEmitter.cpp

```diff
@@ -71,6 +71,143 @@ class ProjMgrYamlCbuildIdx : public ProjMgrYamlBase {
+        modelItem.resolvedPack.pack = packId;
+        model[packId] = modelItem;
+      }
+    }
+  }
+
+  // Stage 3: Add all packs that do not contain wildcard and also not used by any component
+  for (const auto& context : processedContexts) {
+    for (const auto& packItem : context->packRequirements) {
+      // Skip project local packs
+      if (!packItem.path.empty()) {
+        continue;
+      }
+
+      const auto& reqInfo = packItem.pack;
+
+      // Skip wildcard packs
+      if (reqInfo.name.empty() || WildCards::IsWildcardPattern(reqInfo.name)) {
+        continue;
+      }
+
+      const string packId = RtePackage::ComposePackageID(reqInfo.vendor, reqInfo.name, reqInfo.version);
+      if (model.find(packId) == model.end()) {
+        // Pack does not have a version range, so add it
+        ModelItem modelItem;
+        ProjMgrUtils::ConvertToPackInfo(packId, modelItem.info);
+        modelItem.resolvedPack.pack = packId;
+        model[packId] = modelItem;
+      }
+    }
+  }
+
+  // Stage 4: Add all user input expression on the matching resolved pack
+  for (const auto& context : processedContexts) {
+    for (const auto& [userInput, resolvedPacks] : context->userInputToResolvedPackIdMap) {
+      for (const auto& resolvedPack : resolvedPacks) {
+        if (model.find(resolvedPack) != model.end()) {
+          ProjMgrUtils::PushBackUniquely(model[resolvedPack].resolvedPack.selectedBy, userInput);
+        }
+      }
+    }
+  }
+
+  // Stage 5: Process all wildcard patterns from user and add to selected-by list
+  for (const auto& context : processedContexts) {
+    for (const auto& packItem : context->packRequirements) {
+      // Skip project local packs
+      if (!packItem.path.empty()) {
+        continue;
+      }
+
+      const PackInfo& reqInfo = packItem.pack;
+      if (reqInfo.name.empty() || WildCards::IsWildcardPattern(reqInfo.name)) {
+        const string packId = RtePackage::ComposePackageID(reqInfo.vendor, reqInfo.name, reqInfo.version);
+
+        for (auto& [_, item] : model) {
+          if (ProjMgrUtils::IsMatchingPackInfo(item.info, reqInfo)) {
+            ProjMgrUtils::PushBackUniquely(item.resolvedPack.selectedBy, packId);
+          }
+        }
+      }
+    }
+  }
+
+  // Sort model before saving to ensure stable cbuild-pack.yml content
+  vector<pair<string, ModelItem>> sortedModel;
+  for (const auto& item : model) {
+    sortedModel.push_back(item);
+  }
+  std::sort(sortedModel.begin(), sortedModel.end(), [](const pair<string, ModelItem> &item1, const pair<string, ModelItem> &item2) {
+    PackInfo packInfo1, packInfo2;
+    ProjMgrUtils::ConvertToPackInfo(item1.first, packInfo1);
+    ProjMgrUtils::ConvertToPackInfo(item2.first, packInfo2);
+
+    if (packInfo1.vendor == packInfo2.vendor && packInfo1.name == packInfo2.name) {
+      return VersionCmp::Compare(packInfo1.version, packInfo2.version) < 0;
+    }
+    return packInfo1.vendor < packInfo2.vendor || packInfo1.name < packInfo2.name;
```

Comment:
Depending on input data this can lead to `invalid comparator` failing, for example when sorting `ARM::CMSIS` and `Keil::B-U585I-IOT02A_BSP` packs.
From the coverage report it seems there is no test case for this branch:
https://app.codecov.io/gh/Open-CMSIS-Pack/devtools/pull/1143/blob/tools/projmgr/src/ProjMgrYamlEmitter.cpp#L191